### ISSUE TITLE
Accuracy will change the model's output during validation

### ISSUE THREAD

**Issue by qiuxin2012** [intel-analytics/analytics-zoo#2413]
`com.intel.analytics.zoo.pipeline.api.keras.metrics.Accuracy` will change the model's output(in BigDL's Top1Accuracy), if we use a AUC after Top1Accuracy will lead to a wrong AUC score.

**Comment by qiuxin2012** [comment]
https://github.com/intel-analytics/BigDL/blob/4af30dfc73e91cb52efc8c41d7fe76e7a0a96995/spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/ValidationMethod.scala#L192
https://github.com/intel-analytics/BigDL/blob/4af30dfc73e91cb52efc8c41d7fe76e7a0a96995/spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/ValidationMethod.scala#L205
